packages/@aws-cdk/aws-ecs/lib/base/task-definition.ts

```diff
@@ -697,6 +724,24 @@ export class TaskDefinition extends TaskDefinitionBase {
 
     return this.containers.map(x => x.renderContainerDefinition());
   }
+
+  private checkFargateWindowsBasedTasksSize(cpu: string, memory: string, runtimePlatform: RuntimePlatform) {
+    if (Number(cpu) === 1024) {
+      if (Number(memory) < 1024 || Number(memory) > 8192 || (Number(memory)% 1024 !== 0)) {
+        throw new Error(`If define vCPU equal ${cpu}, Memory need in Min. 2GB and Max. 8GB, in 1GB increments, cannot define ${Number(memory)}.`);
```

Comment:
These errors should use the property names that the customer supplied. So vCPU -> cpu and Memory -> memoryMiB. Since the user has to supply the values in MiB, I think it would be easier for the message to us MiB instead of GB. And, change the wording to be more user friendly:

```suggestion
        throw new Error(`If provided cpu is ${cpu}, then memoryMiB must have a min of 1024 and a max of 8192, in 1024 increments. Provided memoryMiB was ${Number(memory)}.`);
```